modules/crawler.py

```diff
@@ -410,240 +345,184 @@ def fetch_shodan_data(ip):
-    except Exception as e:
-        traceback.print_exc()
-    finally:
-        return ipms_indicators
+    # Create the signature
+    ipms_domain_signature_raw = (
+        f"{api_url}/{domain}/{MYIPMS_API_PATH}/timestamp/{timestamp}"
+    )
+    ipms_domain_signature = hashlib.md5(ipms_domain_signature_raw.encode()).hexdigest()
+    # Construct the URL
+    ipms_domain_url = f"{api_url}/{domain}/{MYIPMS_API_PATH}/signature/{ipms_domain_signature}/timestamp/{timestamp}"
+
+    # Repeat the process for the IP address
+    ip_address = socket.gethostbyname(domain)
+    ipms_ip_signature_raw = (
+        f"{api_url}/{ip_address}/{MYIPMS_API_PATH}/timestamp/{timestamp}"
+    )
+    ipms_ip_signature = hashlib.md5(ipms_ip_signature_raw.encode()).hexdigest()
+    ipms_ip_url = f"{api_url}/{ip_address}/{MYIPMS_API_PATH}/signature/{ipms_ip_signature}/timestamp/{timestamp}"
 
+    ipms_indicators.extend(get_ipms_domain_indicators(ipms_domain_url))
+    ipms_indicators.extend(get_ipms_ip_indicators(ipms_ip_url))
+    return ipms_indicators
 
-def get_ipms_domain_indicators(ipms_url):
+@return_empty_if_fails
+def get_ipms_domain_indicators(ipms_url) -> list[Indicator]:
     ipms_indicators = []
     api_result = requests.get(ipms_url)
 
     try:
         data = json.loads(api_result.content)
         if "owners" in data:
             ipms_indicators.append(
-                add_indicator(
+                Indicator(
                     "3-ipms_domain_iprangeowner_cidr", data["owners"]["owner"]["cidr"]
                 )
             )
             ipms_indicators.append(
-                add_indicator(
+                Indicator(
                     "3-ipms_domain_iprangeowner_ownerName",
                     data["owners"]["owner"]["ownerName"],
                 )
             )
             ipms_indicators.append(
-                add_indicator(
+                Indicator(
                     "3-ipms_domain_iprangeowner_address",
                     data["owners"]["owner"]["address"],
                 )
             )
         for dns in data.get("dns", []):
             ipms_indicators.append(
-                add_indicator("3-ipms_domain_nameserver", dns["nameserver"])
+                Indicator("3-ipms_domain_nameserver", dns["nameserver"])
             )
         unique_ips = {entry["ip_address"] for entry in data.get("ip_change_history", [])}
         for ip in unique_ips:
-            ipms_indicators.append(add_indicator("3-ipms_domain_otheripused", ip))
+            ipms_indicators.append(Indicator("3-ipms_domain_otheripused", ip))
 
         return ipms_indicators
 
     except IndexError as e:
-        print(
+        logging.warn(
             "Error hit iterating eunning through IPMS results. Have you hit your IPMS API limit?"
         )
-        traceback.print_exc()
-    except Exception as e:
-        traceback.print_exc()
-    finally:
-        return ipms_indicators
+        raise(e)
```

Comment:
since error handling is now in the wrapper, this now just focuses on the logic and where we do need custom error logging